### ISSUE TITLE
Everything gets converted to Double

### ISSUE THREAD

**Issue by talalryz** [combust/mleap#594]
I serialized a scikit model in python, using a dataset with the following schema, 
TOY_DATA_SCHEMA = StructType(
    [
        StructField("long_feature", LongType()),
        StructField("double_feature", DoubleType()),
        StructField("boolean_feature", BooleanType()),
        StructField("label", BooleanType()),
    ]
)

When I deserialize in scala, and look at `model.inputSchema`, i get the following: 

StructType(List(StructField(long_feature,ScalarType(double,false)), StructField(double_feature,ScalarType(double,false)), StructField(boolean_feature,ScalarType(double,false))),
Map(long_feature -> 0, double_feature -> 1, boolean_feature -> 2),
Map(long_feature -> StructField(long_feature,ScalarType(double,false)), 
double_feature -> StructField(double_feature,ScalarType(double,false)),
 boolean_feature -> StructField(boolean_feature,ScalarType(double,false))

My question is,  "Why is everything being converted to ScalarType(double)?" Is this expected? Or am i doing something wrong

**Comment by ancasarb** [comment]
Hey, could you please provide the example scikit-learn pipeline? It definitely shouldn't be the case that everything gets converted to a double.

**Comment by talalryz** [comment]
@ancasarb 

here is a small example,

This is the dataset
```python
TOY_DATA = [
    (
        i,
        i*11.1 +j,
        1,
        bool(i % 2),
    )
    for i in range(10)
    for j in range(5)
]
TOY_DATA_SCHEMA = StructType(
    [
        StructField("long_feature", LongType()),
        StructField("double_feature", DoubleType()),
        StructField("boolean_feature", BooleanType()),
        StructField("label", BooleanType()),
    ]
)
TOY_DATASET = pd.DataFrame(TOY_DATA, columns=TOY_DATA_SCHEMA.names)
TOY_DATASET_X =  TOY_DATASET[["long_feature", "double_feature", "boolean_feature"]]
TOY_DATASET_Y = TOY_DATASET[["label"]]
```

- Training and serialization

```python
features = ["long_feature", "double_feature", "boolean_feature"]

estimator = LogisticRegression()
estimator.mlinit(
    input_features= "__assembler_features_col",
    prediction_column="label",
)
feature_extractor_tf = FeatureExtractor(
    input_scalars=features, output_vector= "__assembler_features_col"
)

pipeline_model = Pipeline(
    [("feature_extraction", feature_extractor_tf), ("estimator", estimator)]
)
pipeline_model.mlinit()
pipeline_model.fit(X=TOY_DATASET_X, y=TOY_DATASET_Y)
pipeline_model.serialize_to_bundle(path="/work", model_name="mleap_bundle",init=True)
```

- Deserialization and schema printing

```scala
     import ml.combust.mleap.runtime.MleapSupport._
    val file = new File(s"/work/mleap_bundle.zip")
    val transformer = {
      (for(bf <- managed(BundleFile(file))) yield {
        bf.loadMleapBundle().get.root
      }).tried.get
    }
    transformer.inputSchema.fields.foreach(e=> println(e))
```
This is the output i get

```
StructField(long_feature,ScalarType(double,false))
StructField(double_feature,ScalarType(double,false))
StructField(boolean_feature,ScalarType(double,false))
```

**Comment by lucagiovagnoli** [comment]
@ancasarb do you know if this is expected ?

**Comment by ancasarb** [comment]
hi @talalryz @lucagiovagnoli what mleap returns is the required input schema of the pipeline. the feature extractor gets serialized as a vector assembler and in spark that works on Vector(s) of doubles, that's why you are getting all doubles. 

If you nevertheless provide the long, double, boolean and boolean features at scoring time, this won't be an issue as there's some casting available in mleap to make this work, see here https://github.com/combust/mleap/blob/master/mleap-core/src/main/scala/ml/combust/mleap/core/types/Casting.scala. 

please let me know if you have further questions!

**Comment by talalryz** [comment]
I think this is fine
